File: 1511-count-number-of-teams/1511-count-number-of-teams.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        n = len(rating)
        count = 0        
        for i in range(1, n-1):
            smallLeft = 0 
            bigRight = 0
            for j in range(0, i):
                if (rating[j] < rating[i]):
                    smallLeft += 1
            for j in range(i+1, n):
                if (rating[j] > rating[i]):
                    bigRight += 1
            
            count += (smallLeft * bigRight)
            bigLeft = i - smallLeft
            smallRight = n - 1 - i - bigRight
            count += (bigLeft * smallRight)

        return count

File: 1511-count-number-of-teams/1511-count-number-of-teams.java
class Solution {
    public int numTeams(int[] rating) {
        int n = rating.length;
        int count = 0;     
        for (int i = 1; i < n-1; i++) {
            int smallLeft = 0;
            int bigRight = 0;
            
            for (int j = 0; j < i; j++) {
                if (rating[j] < rating[i]) smallLeft += 1;
            }
            for (int j = i+1; j < n; j++) {
                if (rating[j] > rating[i]) bigRight += 1;
            }
            
            count += (smallLeft * bigRight);
            
            int bigLeft = i - smallLeft;
            int smallRight = n - 1 - i - bigRight;
            
            count += (bigLeft * smallRight);
        }

        return count;
    }
}